game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProPurchaseUtils.java

```diff
@@ -90,7 +90,7 @@ public static List<Unit> findMaxPurchaseDefenders(
     final GameState data = proData.getData();
 
     // Determine most cost efficient defender that can be produced in this territory
-    final Resource pus = data.getResourceList().getResource(Constants.PUS);
+    final Resource pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.